aws/resource_aws_cloudwatch_log_stream_test.go

```diff
@@ -65,6 +65,40 @@ func testAccCheckCloudWatchLogStreamDisappears(ls *cloudwatchlogs.LogStream, lgn
 	}
 }
 
+func TestAccAWSCloudWatchLogStream_disappears2(t *testing.T) {
+	var ls cloudwatchlogs.LogStream
+	rName := acctest.RandString(15)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSCloudWatchLogStreamDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSCloudWatchLogStreamConfig(rName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckCloudWatchLogStreamExists("aws_cloudwatch_log_stream.foobar", &ls),
+					testAccCheckCloudWatchLogStreamDisappears2(&ls, rName),
```

Comment:
This new custom function can be replaced with the existing functions from `TestAccAWSCloudWatchLogGroup_disappears()`, e.g.

```go
var lg cloudwatchlogs.LogGroup
// ...
testAccCheckCloudWatchLogGroupExists("aws_cloudwatch_log_group.foobar", &lg),
testAccCheckCloudWatchLogGroupDisappears(&lg),
```

Reply:
I'm less convinced that this is an improvement, but I made the change.